honeyhsc/hsc.go

```diff
@@ -0,0 +1,135 @@
+package honeyhsc
+
+import (
+	"errors"
+	"math"
+
+	"tinygo.org/x/drivers"
+)
+
+var (
+	errSensorMissing = errors.New("hsc: not connected")
+	errDiagnostic    = errors.New("hsc: diagnostic error")
+)
+
+const (
+	measuremask  = drivers.Pressure | drivers.Temperature
+	statusMask   = 0b1100_0000
+	statusOffset = 6
+)
+
+// DevI2C is the TruStability® High Accuracy Silicon Ceramic (HSC) Series is a piezoresistive silicon pressure sensor offering a ratiometric
+// analog or digital output for reading pressure over the specified full scale pressure span and temperature range.
+type DevI2C struct {
+	bus drivers.I2C
+	dev
+	addr uint8
+	buf  [4]byte
+}
+
+func NewDevI2C(bus drivers.I2C, addr uint8, outMin, outMax uint16, pMin, pMax int32) *DevI2C {
```

Comment:
Some I2C devices have 16-bit addresses, although I don't think we have any support for them at the moment. Perhaps it would be a good idea to use `uint16` anyway?

```suggestion
func NewDevI2C(bus drivers.I2C, addr uint16, outMin, outMax uint16, pMin, pMax int32) *DevI2C {
```

Also, can you add some doc comments?